src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -956,10 +977,20 @@ class OpenShiftDSL implements Serializable {
             if ( objectList != null ) {
                 return objectList;
             }
-
+            
             // Otherwise, we need to ask the API server what presently matches
-            OcAction.OcActionResult r = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("get", selectionArgs(), null, "-o=name") );
-            r.failIf( "Unable to retrieve object names: " + this.toString() );
+            OcAction.OcActionResult r = null;
+            ArrayList<String> g = new ArrayList<String>();
+            g.add("get")
+            OcAction.OcActionResult versionCheck = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("help", g, null, null) );
```

Comment:
duh ... yeah, it is an instance var, and a new selector is constructed each time; was distracted by something else, missed the obvious